homeassistant/components/schlage/config_flow.py

```diff
@@ -41,13 +45,34 @@ async def async_step_user(
                 LOGGER.exception("Unknown error")
                 errors["base"] = "unknown"
             else:
-                await self.async_set_unique_id(user_id)
-                return self.async_create_entry(title=username, data=user_input)
+                if not self.reauth_entry:
+                    await self.async_set_unique_id(user_id)
+                    return self.async_create_entry(title=username, data=user_input)
+                self.hass.config_entries.async_update_entry(
+                    self.reauth_entry, data=user_input
+                )
```

Comment:
They shouldn't be able to change the unique id `unique_id` via reauth

Reply:
Ah, good catch. Fixed!